R/rowwiseDT.R

```diff
@@ -13,6 +13,21 @@ rowwiseDT = function(...) {
   nrows = length(body) %/% ncols
   if (length(body) != nrows * ncols)
     stopf("There are %d columns but the number of cells is %d, which is not an integer multiple of the columns", ncols, length(body))
+  is_problematic = vapply_1b(
+    body,
+    function(v) !(is.atomic(v) || is.null(v) || typeof(v) == "list")
+  )
+  if (any(is_problematic)) {
+    first_problem_idx = which(is_problematic)[1L]
+    col_idx = (first_problem_idx - 1L) %% ncols + 1L
+    col_name = header[col_idx]
+    obj_type = typeof(body[[first_problem_idx]])
+    stopf(
+      "In column '%s', received an object of type '%s'.\nComplex objects (like functions, models, etc.) must be wrapped in list() to be stored in a data.table column.\nPlease use `list(...)` for this value.",
```

Comment:
Is this really true? Many `models` are indeed lists, e.g. `lm(mpg~., data=mtcars)`.

Other thing on top of my head would be `expression`s which we do not fully support. Storing environments into a `data.table` seems uncommon to me.